diff --git a/dbus/dbus-connection.c b/dbus/dbus-connection.c
index ac50bd1..6ba75e0 100644
--- a/dbus/dbus-connection.c
+++ b/dbus/dbus-connection.c
@@ -4464,6 +4464,11 @@ _dbus_connection_run_builtin_filters_unlocked_no_update (DBusConnection *connect
   return _dbus_connection_peer_filter_unlocked_no_update (connection, message);
 }
 
+void dbus_connection_dispatch_hook(DBusMessage *message)
+{
+    /* Do nothing, Only as stub function */
+}
+
 /**
  * Processes any incoming data.
  *
@@ -4557,6 +4562,8 @@ dbus_connection_dispatch (DBusConnection *connection)
 
   message = message_link->data;
 
+  dbus_connection_dispatch_hook(message);
+
   _dbus_verbose (" dispatching message %p (%s %s %s '%s')\n",
                  message,
                  dbus_message_type_to_string (dbus_message_get_type (message)),
diff --git a/dbus/dbus-connection.h b/dbus/dbus-connection.h
index 114bd4a..ea59be6 100644
--- a/dbus/dbus-connection.h
+++ b/dbus/dbus-connection.h
@@ -218,6 +218,8 @@ DBusDispatchStatus dbus_connection_get_dispatch_status          (DBusConnection
 DBUS_EXPORT
 DBusDispatchStatus dbus_connection_dispatch                     (DBusConnection             *connection);
 DBUS_EXPORT
+void               dbus_connection_dispatch_hook                (DBusMessage                *message);
+DBUS_EXPORT
 dbus_bool_t        dbus_connection_has_messages_to_send         (DBusConnection *connection);
 DBUS_EXPORT
 dbus_bool_t        dbus_connection_send                         (DBusConnection             *connection,
